src/chains/ethereum/console.log/src/handlers.ts

```diff
@@ -0,0 +1,152 @@
+  // these last 4 bytes of the word that represent the `start`
+  // value.
+  // We add WORD_SIZE to this value because without it it doesn't work.
+  // We add `4` because `memory` includes the function signature, which is 4
+  // bytes, but the `start` position doesn't.
+  const start = memory.readUInt32BE(offset + WORD_SIZE - 4) + WORD_SIZE + 4;
+
+  // the same optimization is applied here.
+  // The word that immediately precedes the data itself is the data's `length`:
+  const length = memory.readUInt32BE(start - 4);
+  return { start, end: start + length };
+}
+
+/**
+ * Ganerates a handler function for solidity's bytes1-bytes32 types
+ * @param length From 1 to 32.
+ * @returns A function that accepts a memory Buffer and an offset.
+ */
+export function fixedBytes(length: number) {
+  return (memory: Buffer, offset: number) =>
+    handleBytes(memory, offset, offset + length);
+}
+
+/**
+ * Returns the hex representation of the bytes in `memory`.
+ *
+ * @param memory
+ * @param start The start of the data in the given memory.
+ * @param end The end of the data in the given memory (not inclusive).
+ * @returns
+ */
+function handleBytes(memory: Buffer, start: number, end: number) {
+  return `0x${memory.subarray(start, end).toString("hex")}`;
+}
+
+export const int256 = (memory: Buffer, offset: number) => {
+  // convert from two's compliment to signed BigInt
+  const twosComplementBuffer = memory.subarray(offset, offset + WORD_SIZE);
+  const twosComplementBigInt = bufferToBigInt(twosComplementBuffer);
+  if (twosComplementBuffer[0] & 128) {
+    // if the first bit is `1` we need to convert from two's compliment
+    return twosComplementBigInt | OR_WITH_TWOS_COMPLEMENT;
+  } else {
+    // if the first bit is not `1` we can treat it as unsigned
+    return twosComplementBigInt;
+  }
+};
+
+/**
+ * Handle decoding of solidity's `uint256` type from EVM memory
+ * @param memory
+ * @param offset
+ * @returns
+ */
+export const uint256 = (memory: Buffer, offset: number) =>
+  bufferToBigInt(memory.subarray(offset, offset + WORD_SIZE));
+
+/**
+ * Handle decoding of solidity's dynamic `string` type from EVM memory
+ * @param memory
+ * @param offset
+ * @returns
+ */
+export const string = (memory: Buffer, offset: number) => {
+  const { start, end } = getDynamicDataMarkers(memory, offset);
+  return memory.toString("utf8", start, end);
+};
+
+/**
+ * Handle decoding of solidity's `bool` type from EVM memory
+ * @param memory
+ * @param offset
+ * @returns
+ */
+export const bool = (memory: Buffer, offset: number) =>
+  memory[offset + 31] !== 0;
+
+/**
+ * Handle decoding of solidity's `address` type from EVM memory
+ * @param memory
+ * @param offset
+ * @returns A 20 byte hex encoding string, prefixed with 0x
+ */
+export const address = (memory: Buffer, offset: number) =>
+  `0x${memory
+    .subarray(offset + 12, offset + WORD_SIZE)
+    .toString("hex")
+    .padStart(40, "0")}`;
```

Comment:
Why do we need to `padStart`?  Won't this always be 20 bytes converted to a hex string?

Reply:
I'm testing removing the `padStart` locally now.

Reply:
I _can_ think of a way to make things get weird here by executing a malformed `STATICCALL` manually (outside of our console.sol), but I don't think that is real concern here. Good find!